server/app/filters/NonUserRoutePrefixes.java

```diff
@@ -0,0 +1,41 @@
+package filters;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import play.mvc.Http;
+
+/**
+ * Path prefixes for HTTP routes that do not involve a user session and can be safely skipped by
+ * filters that act on the user session or account.
+ */
+enum NonUserRoutePrefixes {
+  API("/api/"),
+  ASSETS("/assets/"),
+  DEV("/dev/"),
+  FAVICON("/favicon"),
+  PLAY_INDEX("/playIndex"),
+  METRICS("/metrics");
+
+  public static ImmutableList<String> ALL =
+      Arrays.stream(NonUserRoutePrefixes.values())
+          .map(NonUserRoutePrefixes::getPrefix)
+          .collect(ImmutableList.toImmutableList());
+
+  public static boolean anyMatch(Http.RequestHeader requestHeader) {
+    return ALL.stream().anyMatch(prefix -> requestHeader.path().startsWith(prefix));
+  }
+
+  public static boolean noneMatch(Http.RequestHeader requestHeader) {
+    return !anyMatch(requestHeader);
+  }
```

Comment:
**suggestion (non-blocking):** Considering using `stream().noneMatch()` here. I tend to trust the libraries to do the right thing.

Reply:
I had it that way and changed it since that duplicated the lambda. Could also factor the lambda out into another method, which just seemed like needless indirection for two simple methods like this.